### DISCUSSION TITLE
Dependency injection error

### DISCUSSION THREAD

**Discussion by pcmanprogrammeur** [symfony/symfony#49790] [Q&A] [answered]
Hi,
I am using the dependency injection between service and i have the error :
`Maximum function nesting level of '256' reached, aborting`

Here is the simple sample :
```
class UserService
{
    public function __construct(ResumeService $resumeService) {
    }
}
```

```
class ResumeService
{
    public function __construct(UserService $userService){ }
}
```

Do you know how i can fix this problem please ?

**Comment by ahundiak** [comment, marked as answer]
This is what is known as a circular dependency.  You can search `symfony service circular dependency` for strategies to resolve.  

Marking one service as [lazy](https://symfony.com/doc/current/service_container/lazy_services.html#configuration) will often fix the problem.  If not then a bit of refactoring might be needed.

**Reply by pcmanprogrammeur** [reply]
Thank you for the lazy parameter, it works perfectly !

**Comment by ghost** [comment]
It's obvious that you get this error because you can't create two services that depend on each other. First tries to create second, but second depends on first so your're in the infinite loop.
